File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        int len = s.length();

        for(int i = 0; i < len; i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '['){
                st.push(s[i]);
            }else{
                if(st.empty()) return false;

                else if(s[i] == ')'){
                    if(st.top() == '(') st.pop();
                    else return false;
                }else if(s[i] == '}'){
                    if(st.top() == '{') st.pop();
                    else return false;
                }else{
                    if(st.top() == '[')st.pop();
                    else return false;
                }
            }
        }
        return st.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        mapp = {
            ')' : '(',
            '}' : '{',
            ']' : '['
        }
        stack = []
        for val in s:
            if val in mapp.values():
                stack.append(val)
            else:
                if not stack or stack[-1] != mapp[val]:
                    return False
                stack.pop()
        
        return len(stack) == 0